### DISCUSSION TITLE
Getting "Failed to authorize: Null intent received"

### DISCUSSION THREAD

**Discussion by antoniodourado** [Bdaya-Dev/oidc#28] [Q&A]
I'm haveing this issue after I authorize the login within my browser. There is no underscore or uppercase in my project. Anyone to help me out?

**Comment by ahmednfwela** [comment]
Do you get this in android/ios or web ?

**Reply by antoniodourado** [reply]
Android/iOS. The issue was on the server. However, I'm having another problem.

The server is returning for the JsonWebKeySet a List of Maps as does this example from Google: https://www.googleapis.com/oauth2/v3/certs

However, in line 518 I'm getting an error that JsonWebKeySet.fromJson expects a Map<String, dynamic> not a List<dynamic>, which is in fact a List<Map<String,dynamic>>:

`type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>'`

Not sure if it's related, but the id_token is getting a severe message that it failed to verify and then that audiences doesn't contain my clientId.

`I/flutter ( 5370): SEVERE: 2023-10-22 17:37:42.562633: Failed to verify id_token, using unverified instead.
I/flutter ( 5370): WARNING: 2023-10-22 17:37:42.567964: found a JWT, but failed the validation test: JoseException: Audiences does not contain clientId 'myclientid'.`

**Reply by ahmednfwela** [reply]
seems like yet another problem with jose https://github.com/Bdaya-Dev/oidc/issues/9
what openid provider are you using? is it spec compliant ?

**Reply by ahmednfwela** [reply]
Thanks, I have seen the discovery document, and I see the problem now.
the jwks uri used in your configuration returns this:
```json
[
  {
    "n": "xxx",
    "e": "xxx",
    "kty": "xxx",
    "kid": "xxx",
    "alg": "RS256",
    "use": "sig"
  }
]
```
but it should be:
```json
{
"keys": [
  {
    "n": "xxx",
    "e": "xxx",
    "kty": "xxx",
    "kid": "xxx",
    "alg": "RS256",
    "use": "sig"
  }
]
}
```

**Reply by ahmednfwela** [reply]
Please check https://datatracker.ietf.org/doc/html/rfc7517#section-5

> A JWK Set is a ***JSON object*** that represents a set of JWKs.  The JSON
>    object MUST have a ***"keys"*** member, with its value being an array of
>    JWKs.  This JSON object MAY contain whitespace and/or line breaks.

**Reply by antoniodourado** [reply]
Thank you! I'll ask for the server team to make implementation and I'll let you know if the issue persists!

**Reply by ahmednfwela** [reply]
You are welcome! I am also available on email if you don't want to share any private information on github:
ahmednfwela@bdaya-dev.com

**Comment by antoniodourado** [comment]
It seems I got the auth process. However, I'm getting the following issues:

1. `Failed to verify id_token, using unverified instead.` - We are using public client, is it the reason?
2. `can't monitor user session due to lack of sessionState (null) or checkSessionIframe (null)` - I see that we don't have the Iframe on discovery, but how about the session state?
3. When I authorize with the server, it's being shown two instances of the same app to open the redirect link. One of them I get the original error message of null intent but get redirected and the other I manage to auth/get user but I'm not redirected and get the messages 1 and 2.

Any ideas?

**Reply by ahmednfwela** [reply]
1. there is a reason that makes the verification part fail, which is stated above in your original comment `Audiences does not contain clientId 'myclientid'.` , when your auth server issues an id_oken, it MUST contain an `aud` claim equal to your client id (`myclientid` in this case).
2. to monitor the session state, you have 2 choices:
    2.1. Use the state management spec, which requires a check session iframe in the discovery document, and a session state in the authorization response (NOTE this is only supported on web).  
    2.2. Change the refresh token duration to be low (10 seconds maybe), when the user logs out and the refresh token is revoked, they will fail the next token refresh.
3. Do you perhaps call login method multiple times ?

**Reply by ahmednfwela** [reply]
I should state that reducing the refresh token duration isn't a good option when scaling for large amounts of users.

**Reply by antoniodourado** [reply]
The verification is failing due another issue on the auth server and their team is solving it.

Regarding the login method, it is not being called multiple times. It is very weird.

**Reply by ahmednfwela** [reply]
I am thinking it may be a bug on the server that redirects to the same page in a new tab? but maybe i am wrong

**Reply by antoniodourado** [reply]
It seems that removing the intent filter in AndroidManifest.xml redirected from the browser to my app. However the route is completely ignored. Should I work on it manually throught redirect from GoRoute?

**Reply by ahmednfwela** [reply]
you don't need to use intent filters if you add the gradle entry as explained in the setup document
https://bdaya-dev.github.io/oidc/oidc-getting-started/#appauth-android-setup

going back to the original url is a different problem that depends on your routing package, and is solved in the example here 

https://github.com/Bdaya-Dev/oidc/blob/f73bda84cc9e12fef9fd27549c31cd2259e7adc2/packages/oidc/example/lib/main.dart#L33-L70

where secret page represents a page that should be guarded by auth.
